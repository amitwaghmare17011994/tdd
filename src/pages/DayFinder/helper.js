import moment from 'moment'

const DAYS = {
    0: 'Sunday',
    1: 'Monday',
    2: 'Tuesday',
    3: 'Wednesday',
    4: 'Thursday',
    5: 'Friday',
    6: 'Saturday'
}


export const findDay = (dateObject) => {
    const date = moment(dateObject, "DD/MM/YYYY")
    const dayOfMonth = date.date()
    const year = date.year().toString()
    const month = date.month() + 1
    let lastTwoDigitOfYear = parseInt(year[year.length - 2] + year[year.length - 1])
    const firstTwoDigitOfYear = parseInt(year[0] + year[1])
    let monthNumber = month - 2
    if (monthNumber <= 0) {
        monthNumber = monthNumber === 0 ? 12 : 11
        lastTwoDigitOfYear--
    }


    const resIndex = dayOfMonth + parseInt(parseInt(13 * monthNumber - 1) / 5) + lastTwoDigitOfYear + parseInt(lastTwoDigitOfYear / 4) + parseInt(firstTwoDigitOfYear / 4) - 2 * firstTwoDigitOfYear
    const offset = resIndex % 7
    return DAYS[offset]

}