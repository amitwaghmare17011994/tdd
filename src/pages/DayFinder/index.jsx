import React, { useState } from 'react'
import { Button, Paper, TextField, Typography } from '@material-ui/core'
import LayoutContainer from '../../components/LayoutContainer'
import { findDay } from './helper'

function DayFinder() {
    const [state, setState] = useState({
        date: '',
        res: ''
    })
    const onChange = (e) => {

        setState({ ...state, date: e.target.value })


    }
    const onCaclulate = () => {
        const day = findDay(state.date)
        setState({ ...state, res: day })
    }
    
    return (
        <LayoutContainer>
            <Paper elevation={3} style={{ height: 500, width: 550, display: 'flex', flexDirection: 'column', justifyContent: 'space-evenly', alignItems: 'center' }}>
                <TextField onChange={onChange} placeholder={'Enter Date'} />
                <Button onClick={onCaclulate} color={'primary'} variant={'contained'}>
                    Calculate
                </Button>
                <Typography>
                    {state.res}
                </Typography>
            </Paper>
        </LayoutContainer>
    )
}

export default DayFinder
