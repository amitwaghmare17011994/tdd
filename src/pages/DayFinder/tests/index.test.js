
import { Button, Typography } from '@material-ui/core';
import { create, act } from 'react-test-renderer';
import DayFinder from '..';


describe('Day Finder', () => {
    let root;
    let instance;

    beforeEach(() => {
        root = create(<DayFinder />)
        instance = root.root
    })


    it('should render Day Finder', () => {
        expect(root.toJSON()).toMatchSnapshot();
    })

    it('should find day from 17/01/1994 (monday)', () => {
        const textField = instance.findByProps({ placeholder: 'Enter Date' })
        const button = instance.findByType(Button)
        act(() => {
            const date = '17/01/1994'
            textField.props.onChange({ target: { value: date } })
        })
        act(() => {
            button.props.onClick()
        })
        const res = instance.findByType(Typography)
        expect(res.props.children).toBe('Monday')


        act(() => {
            const date = '02/05/2021'
            textField.props.onChange({ target: { value: date } })
        })
        act(() => {
            button.props.onClick()
        })

        expect(res.props.children).toBe('Sunday')


        act(() => {
            const date = '27/02/2023'
            textField.props.onChange({ target: { value: date } })
        })
        act(() => {
            button.props.onClick()
        })

        expect(res.props.children).toBe('Monday')



    })
})