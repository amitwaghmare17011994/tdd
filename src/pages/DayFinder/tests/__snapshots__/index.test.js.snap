// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Finder should render Day Finder 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <div
      className="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": 500,
          "justifyContent": "space-evenly",
          "width": 550,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter Date"
            required={false}
            type="text"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Calculate
        </span>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        
      </p>
    </div>
  </div>
</div>
`;
